src/ChoiceScreen.tsx

```diff
@@ -1,64 +1,91 @@
+  };
 
-    handleDisplayNameUpdate = (event:React.ChangeEvent<HTMLInputElement>) =>{
-        this.setState({displayName:event.target.value});
-    }
+  handleDisplayNameUpdate = (event: React.ChangeEvent<HTMLInputElement>) => {
+    this.setState({ displayName: event.target.value });
+  };
 
-    handleJoinRoom = (e:React.MouseEvent<HTMLButtonElement>) =>{
-        const isValid = this.validateJoinRoom();
-        if(isValid){
-            this.props.handleJoinRoom(this.state.roomName,this.state.displayName);
-        }
+  emptyDisplayName = () => {
+    let colour = '';
+    const isValid = this.validateCreateRoom();
+    if (!isValid) {
+      colour = '2px solid red';
     }
+    return this.setState({ borderColorDisplayName: colour });
+  };
 
-    handleCreateRoom = (e:React.MouseEvent<HTMLButtonElement>) =>{
-        const isValid = this.validateCreateRoom();
-        if(isValid){
-            this.props.handleCreateRoom(this.state.displayName);
-        } 
-    }
+  emptyJoinName = () => {
+    return this.setState({ borderColorRoomName: '2px solid red' });
+  };
 
-    validateJoinRoom =() =>{   
-        return this.state.displayName.length > 0 && this.state.roomName.length > 0;
+  handleJoinRoom = (e: React.MouseEvent<HTMLButtonElement>) => {
+    const isValid = this.validateJoinRoom();
+    if (isValid) {
+      this.props.handleJoinRoom(this.state.roomName, this.state.displayName);
+    } else {
+      this.emptyJoinName();
+      this.emptyDisplayName();
     }
+  };
 
-    validateCreateRoom =() =>{   
-        return this.state.displayName.length > 0;
+  handleCreateRoom = (e: React.MouseEvent<HTMLButtonElement>) => {
+    const isValid = this.validateCreateRoom();
+    if (isValid) {
+      this.props.handleCreateRoom(this.state.displayName);
+    } else {
+      this.emptyDisplayName();
     }
+  };
 
-    render(){
-        const selectionComponent = (<div>
-            <label>
-                Display Name: 
-            </label>
-            <input type="text" name="displayName" onChange={this.handleDisplayNameUpdate}/>
-            <br/>
-            <button type='button'onClick={this.handleJoinRoom}>Join Room</button>
-            <input type="text" name="roomCode" onChange={this.handleRoomNameUpdate}/>
-            <button onClick={this.handleCreateRoom}>Create Room</button>
-            </div>
-        )
+  validateJoinRoom = () => {
+    return this.state.displayName.length > 0 && this.state.roomName.length > 0;
+  };
 
-        return(
-            <div>
-            {selectionComponent}
-            </div>
-        )
-    };
+  validateCreateRoom = () => {
+    return this.state.displayName.length > 0;
+  };
+
+  render() {
+    const selectionComponent = (
+      <div>
+        <label>Display Name:</label>
+        <input
+          type='text'
+          name='displayName'
+          onChange={this.handleDisplayNameUpdate}
+          style={{ border: this.state.borderColorDisplayName }}
```

Comment:
Will the colour of the box turn back to normal once the user starts to type in it?